code/game/objects/items/weapons/null_rod.dm

```diff
@@ -266,6 +266,42 @@
 	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/swords_axes.dmi', "right_hand" = 'icons/mob/in-hand/right/swords_axes.dmi')
 	fluff_pickup = "smite"
 
+/obj/item/weapon/nullrod/mosinnagant
+	name = "mosin nagant"
+	desc = "Many centuries later, it's still drenched in cosmoline, just like the Murdercube intended. This one cannot be fired."
+	icon = 'icons/obj/gun.dmi'
+	icon_override = "nagant"
+	icon_state = "nagant"
+	item_state = "nagant"
+	slot_flags = SLOT_BELT | SLOT_BACK
+	w_class = W_CLASS_LARGE
+	attack_verb = list("bashes", "smashes", "buttstrokes")
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/guninhands_left.dmi', "right_hand" = 'icons/mob/in-hand/right/guninhands_right.dmi')
+
+/obj/item/weapon/nullrod/mosinnagant/attackby(var/obj/item/A as obj, mob/living/user as mob)
+	..()
+	if(istype(src, /obj/item/weapon/nullrod/mosinnagant/obrez))
```

Comment:
No need for this, just have the obrez's attackby contain `if (istype(A, /obj/item/weapon/circular_saw) return`.